### PR TITLE
fix: client/server SSL settings fail when 'ssl.key.password' is set

### PR DESCRIPTION
**Pull request by spena**
### Description 
Fixes https://github.com/confluentinc/ksql/issues/6719

Different keystore and key passwords are not supported in Vert.x API. Internally. Vert.x always wants to decrypt the Private key using the keystore password. To fix this, this PR decrypts the Private key with the password specified in `ssl.key.password`, then re-encrypts it with the keystore password. The resulted encrypted bytes are then passed to Vert.x to handle the SSL communication.

The code has the re-encryption part in `KeyStoreUtil.java` already. I  just re-use the same keystore password when encrypting the key again.

The rest of the code in `Server.java`, `KsqlClient.java` and `InternalKsqlClientFactory.java` look for the `ssl.key.password` config, and passes the key password to the `KeyStoreUtil`. This is supported only when the keystore is a `JKS` type and when a key alias is found in the configs too. FYI, 'PKCS12` does not support a keystore with different key password (both must be the same).

A new client config `ssl.keystore.alias` is added so that clients can configure the `ssl.key.password`.

### Testing done 
Added unit tests
Verified manually.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Review by JimGalasyn — APPROVED** [review]
Great, thanks!

**Comment by spena** [comment]
Hold on on this review until the following refactor is merged: https://github.com/confluentinc/ksql/pull/6823

**Comment by spena** [comment]
Ready for review. https://github.com/confluentinc/ksql/pull/6823 is merged.

**Review by JimGalasyn — APPROVED** [review]
LGTM!

**Review by agavra — APPROVED** [review]
LGTM! Given that this is security related (and my knowledge is pretty surface level) you may or may not decide that you want another reviewer as well.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestConfig.java
Comment (spena) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestConfig.java:422]: Ah yes, Done.

**Pull request merged**
